### ISSUE TITLE
Unable to create tenant registration but only if tenant already requested unsuccessfully

### ISSUE THREAD

**Issue by jason-fox** [ScorpioBroker/ScorpioBroker#508]
Scorpio 4.1.11. - Empty Context Broker, no data.

Start by requesting data from a non-existing tenant:

```console
curl -L 'http://localhost:9090/ngsi-ld/v1/csourceRegistrations/?type=FillingLevelSensor' \
-H 'Accept: application/ld+json' \
-H 'Link: <http://context/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
-H 'NGSILD-Tenant: openiot'
```

The response **correctly** states the tenant does not exist at this point.

```json
{
  "type":"https://uri.etsi.org/ngsi-ld/errors/TenantNotFound",
  "title":"Tenant not found.",
  "detail":{
       "message":"openiot tenant was not found"
   },
  "errorCode":404
}
```

Now attempt to create a registration on that tenant:

```console
curl  -iX POST -L 'http://localhost:1026/ngsi-ld/v1/csourceRegistrations/' \
-H 'Content-Type: application/json' \
-H 'Link: <http://context/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
-H 'NGSILD-Tenant: openiot' \
-d '   {
        "type": "ContextSourceRegistration",
        "contextSourceInfo": {
            "key": "jsonldContext",
            "value": "http://context/ngsi-context.jsonld"
        },
        "endpoint": "http://iot-agent:4041",
        "information": [
            {
                "entities": [
                    {
                        "id": "urn:ngsi-ld:Device:filling111",
                        "type": "FillingLevelSensor"
                    }
                ],
                "propertyNames": [
                    "add",
                    "remove",
                    "refill"
                ]
            }
        ],
        "mode": "exclusive",
        "operations": "updateOps"
 }'
```   

Response:

```json
{
    "type":"https://uri.etsi.org/ngsi-ld/errors/InternalError",
    "title":"Internal error",
     "detail":{
          "message":null
     },
    "errorCode":500
}
```

Stack trace

<summary>

<details>

```text
024-01-11 10:08:57 eu.neclab.ngsildbroker.commons.exceptions.ResponseException
2024-01-11 10:08:57     at eu.neclab.ngsildbroker.commons.storage.ClientManager.lambda$findDataBaseNameByTenantId$8(ClientManager.java:163)
2024-01-11 10:08:57     at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
2024-01-11 10:08:57     at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:68)
2024-01-11 10:08:57     at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
2024-01-11 10:08:57     at io.smallrye.mutiny.vertx.AsyncResultUni.lambda$subscribe$1(AsyncResultUni.java:35)
2024-01-11 10:08:57     at io.smallrye.mutiny.vertx.DelegatingHandler.handle(DelegatingHandler.java:25)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
2024-01-11 10:08:57     at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
2024-01-11 10:08:57     at io.vertx.sqlclient.impl.QueryResultBuilder.tryComplete(QueryResultBuilder.java:88)
2024-01-11 10:08:57     at io.vertx.sqlclient.impl.QueryResultBuilder.tryComplete(QueryResultBuilder.java:32)
2024-01-11 10:08:57     at io.vertx.core.Promise.complete(Promise.java:66)
2024-01-11 10:08:57     at io.vertx.core.Promise.handle(Promise.java:51)
2024-01-11 10:08:57     at io.vertx.core.Promise.handle(Promise.java:29)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
2024-01-11 10:08:57     at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
2024-01-11 10:08:57     at io.vertx.core.impl.future.Transformation$1.onSuccess(Transformation.java:63)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureImpl.addListener(FutureImpl.java:196)
2024-01-11 10:08:57     at io.vertx.core.impl.future.PromiseImpl.addListener(PromiseImpl.java:23)
2024-01-11 10:08:57     at io.vertx.core.impl.future.Transformation.onSuccess(Transformation.java:44)
2024-01-11 10:08:57     at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
2024-01-11 10:08:57     at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
2024-01-11 10:08:57     at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
2024-01-11 10:08:57     at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
2024-01-11 10:08:57     at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:413)
2024-01-11 10:08:57     at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
2024-01-11 10:08:57     at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2024-01-11 10:08:57     at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2024-01-11 10:08:57     at java.base/java.lang.Thread.run(Thread.java:1570)
```

</details>

</summary>

**Comment by jason-fox** [comment]
Note that this error does not occur if the Tenant has not been queried prior to creating the Registration.

**Comment by jason-fox** [comment]
Fixed 4.1.15
